src/Microsoft.ML.Data/Utilities/StreamUtils.cs

```diff
@@ -174,5 +174,40 @@ private static string[] Expand(string pattern)
             return matchList.ToArray();
         }
 #endif
+
+        public static string ReadEntry(this TextReader sr)
+        {
+            string strReturn = string.Empty;
+
+            // get first bit
+            strReturn += sr.ReadLine();
+
+            // And get more lines until the number of quotes is even
+            while (strReturn.GetNumberOf("\"").IsOdd())
+            {
+                string strNow = sr.ReadLine();
+                strReturn += strNow;
+            }
+
+            // Then return what we've gotten
+            if (strReturn == string.Empty)
+            {
+                return null;
+            }
+            else
+            {
+                return strReturn;
+            }
+        }
+
+        public static int GetNumberOf(this string s, string strSearchString)
+        {
+            return s.Length - s.Replace(strSearchString, string.Empty).Length;
```

Comment:
Interesting approach :)

Reply:
Actually there's a bug.... It should divided by the length of `strSearchString` to get the correct result. also if I do that, it won't work on string.Empty (div by 0)... so hard to write the correct code!